gc/gc-base/src/main/java/org/projectnessie/versioned/persist/gc/DistributedIdentifyUtil.java

```diff
@@ -0,0 +1,278 @@
+                        reference))
+            .collect();
+    return mergeLiveContentResults(bloomFilterMaps);
+  }
+
+  public IdentifiedResult getIdentifiedResults(
+      Map<String, ContentBloomFilter> liveContentsBloomFilterMap, List<Reference> references) {
+
+    List<IdentifiedResult> results =
+        new JavaSparkContext(session.sparkContext())
+            .parallelize(references, references.size())
+            .map(
+                reference ->
+                    computeExpiredContents(
+                        clientBuilderConfiguration, liveContentsBloomFilterMap, reference))
+            .collect();
+    // TODO: write each task output directly into the spark table.
+    IdentifiedResult identifiedResult = new IdentifiedResult();
+    results.forEach(result -> identifiedResult.merge(result.getContentValues()));
+    return identifiedResult;
+  }
+
+  protected Map<String, ContentBloomFilter> computeLiveContents(
+      Map<String, String> clientBuilderConfig,
+      Map<String, Instant> cutOffTimestampPerRef,
+      Instant defaultCutOffTimestamp,
+      Reference reference) {
+    NessieApiV1 api = getApi(clientBuilderConfig);
+    Instant cutOffTimestamp =
+        cutOffTimestampPerRef.get(reference.getName()) == null
+            ? defaultCutOffTimestamp
+            : cutOffTimestampPerRef.get(reference.getName());
+    return walkLiveReference(
+        api,
+        reference,
+        commitMeta ->
+            // If the commit time is newer than (think: greater than or equal to) cutoff-time, then
+            // commit is live.
+            Objects.requireNonNull(commitMeta.getCommitTime()).compareTo(cutOffTimestamp) >= 0);
+  }
+
+  protected @NotNull IdentifiedResult computeExpiredContents(
+      Map<String, String> clientBuilderConfig,
+      Map<String, ContentBloomFilter> liveContentsBloomFilterMap,
+      Reference reference) {
+    NessieApiV1 api = getApi(clientBuilderConfig);
+    return walkAllReference(api, reference, liveContentsBloomFilterMap);
+  }
+
+  private Map<String, ContentBloomFilter> mergeLiveContentResults(
+      List<Map<String, ContentBloomFilter>> bloomFilterMaps) {
+    Map<String, ContentBloomFilter> output = new HashMap<>();
+    bloomFilterMaps.forEach(
+        map ->
+            map.forEach(
+                (k, v) -> {
+                  if (output.containsKey(k)) {
+                    output.get(k).merge(v);
+                  } else {
+                    output.put(k, v);
+                  }
+                }));
+    return output;
+  }
+
+  private Map<String, ContentBloomFilter> walkLiveReference(
+      NessieApiV1 api, Reference reference, Predicate<CommitMeta> liveCommitPredicate) {
+    final Logger LOG = LoggerFactory.getLogger(DistributedIdentifyUtil.class);
+    Map<String, ContentBloomFilter> bloomFilterMap = new HashMap<>();
+    Set<ContentKey> deletedKeys = new HashSet<>();
+    Consumer<LogResponse.LogEntry> perCommit =
```

Comment:
Does this need to be declared here? It's confusing to declare a `Consumer` here, just delegate to another function but use it 40 lines below.